### PR TITLE
feat(OnyxErrorTooltip): use onyx tooltip for errors on checkbox/radio/switch

### PR DESCRIPTION
**Pull request by BoppLi**
Relates to #1600 
- create OnyxErrorTooltip component: wraps a given component inside a `danger` OnyxTooltip to show an error message if it exists, or returns the component untouched elsewise.
- use OnyxErrorTooltip instead of `title` to show error Messages for
  - OnyxCheckbox / OnyxCheckboxGroup
  - OnyxRadioButton / OnyxRadioGroup
  - OnyxSwitch
- only shows the error tooltip on `:user-invalid`

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxCheckboxGroup/OnyxCheckboxGroup.stories.ts [hunk 1]
```diff
@@ -83,6 +92,7 @@ export const Disabled = {
 /**
  * A checkbox group with long labels that will be truncated.
  * You can set the "truncation" property of the options to choose between the different truncation types.
+ * TODO: broken because of the tooltip
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxCheckboxGroup/OnyxCheckboxGroup.stories.ts:95]: todo - fix truncation

File: packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// TODO: should this component be exported for onyx users?
+
+const props = defineProps<{
+  /**
+   * The given component will be shown inside a tooltip when
+   * errorMessages are provided. Without errorMessages, the
+   * component will not be rendered inside a slot.
+   */
+  errorMessages?: FormErrorMessages;
+  // TODO: clarify if this feature is wanted
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue:15]: todo discuss in daily

File: packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<script lang="ts" setup>
+import { computed, ref } from "vue";
+import { type FormErrorMessages, getCustomErrorText } from "../../composables/useCustomValidity";
+import OnyxTooltip from "../OnyxTooltip/OnyxTooltip.vue";
+
+// TODO: should this component be exported for onyx users?
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue:6]: todo discuss in daily

File: packages/sit-onyx/playwright/snapshots/components/OnyxCheckboxGroup/CheckboxGroup-webkit-linux.png
Comment (BoppLi) [inline packages/sit-onyx/playwright/snapshots/components/OnyxCheckboxGroup/CheckboxGroup-webkit-linux.png:1]: TODO fix missing "Select all" divider line

File: packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue (hunk 3 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue:6]: don't export it, and no storybook file

File: packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<script lang="ts" setup>
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue:1]: todo component test

File: packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue (hunk 2 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue:15]: it does not :)

File: packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.ct.tsx [hunk 5]
```diff
@@ -69,28 +69,44 @@ test.describe("Screenshot tests", () => {
...
+          value="test-value"
+        />
+      );
+    },
     beforeScreenshot: async (component, page, column, row) => {
+      if (row !== "default") {
+        // add space for tooltip
+        await component.evaluate((element) => {
+          element.style.paddingTop = "3rem";
+        });
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.ct.tsx:96]: Is there a reason why you don't set this as `style` attribute above?

File: packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.stories.ts [hunk 6]
```diff
@@ -56,6 +56,15 @@ export const Required = {
     ...Default.args,
     required: true,
   },
+  decorators: [
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.stories.ts:59]: Hmm the example looks a little off when the tooltip is not shown
image

File: packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.ct.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+import OnyxErrorTooltip from "./OnyxErrorTooltip.vue";
+
+test("should render without error-tooltip", async ({ mount, makeAxeBuilder }) => {
+  // ARRANGE
+  const component = await mount(<OnyxErrorTooltip>Dummy content</OnyxErrorTooltip>);
+
+  // ACT
+  await component.getByText("Dummy content").hover();
+  // ASSERT
+  await expect(component.getByRole("tooltip")).toHaveCount(0);
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.ct.tsx:11]:
```suggestion
  await expect(component.getByRole("tooltip")).toBeHidden();
```

Isn't this more appropriate?

File: packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    <!-- sends the given component to the desired target without destroying the component -->
+    <Teleport :disabled="!targetRef" :to="targetRef">
+      <slot></slot>
+    </Teleport>
+  </div>
+</template>
+
+<style lang="scss">
+.onyx-error-tooltip {
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue:55]: We should use the CSS layers mixin here

File: packages/sit-onyx/src/components/OnyxErrorTooltip/types.ts
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxErrorTooltip/types.ts:1]: File seems to be unsed

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.stories.ts (hunk 6 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.stories.ts:59]: as discussed, I added a point about that in 
- #730
:)

File: packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue (hunk 8 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.vue:55]: dammit, still missing in my emmet setup :D

File: packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.ct.tsx (hunk 7 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxErrorTooltip/OnyxErrorTooltip.ct.tsx]: for me, "Element is not visible" sounded like it exists but is visually hidden, I thought to have count sounds more like "does not exist". 
¯\_(ツ)_/¯

I updated it toBeHidden if we use that more frequently

File: packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.ct.tsx (hunk 5 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.ct.tsx]: I worried it might impact the tooltip, but it doesn't - all good :)

**Review by larsrickert — APPROVED** [review]

**Pull request merged**